Packages/com.unity.inputsystem/InputSystem/Actions/InputActionSetupExtensions.cs

```diff
@@ -487,12 +487,7 @@ private static int AddBindingInternal(InputActionMap map, InputBinding binding,
             else
                 ArrayHelpers.InsertAt(ref map.m_Bindings, bindingIndex, binding);
 
-            // Invalidate per-action binding sets so that this gets refreshed if
-            // anyone queries it.
-            map.ClearPerActionCachedBindingData();
-
-            // Make sure bindings get re-resolved.
-            map.LazyResolveBindings();
+            InvalidateActionMap(map);
```

Comment:
This will have to get called from several more places. Probably each one in `InputActionSetupExtensions` and `InputActionRebindingExtensions` which does the `ClearPerActionCachedBindingData` and `LazyResolveBindings` call pattern here.

Reply:
Hmm, now I think I shouldn't have combined these calls into this method. We probably don't want to mark an asset as dirty in all cases where we use call ClearPerActionCachedBindingData and LazyResolveBindings. For example, ApplyBindingOverride does nothing to the asset that would require it to be reloaded on exit. I've undone this change and we can add a refactoring task later if you still want to combine the other two method calls.